cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -247,13 +266,29 @@ func (g *gitHubRepository) getReleaseByTag(tag string) (*github.RepositoryReleas
 
 	client := g.getClient()
 
-	release, _, err := client.Repositories.GetReleaseByTag(context.TODO(), g.owner, g.repository, tag)
-	if err != nil {
-		return nil, g.handleGithubErr(err, "failed to read release %q", tag)
-	}
+	var release *github.RepositoryRelease
+	var retryError error
+	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+		var getReleasesErr error
+		release, _, getReleasesErr = client.Repositories.GetReleaseByTag(context.TODO(), g.owner, g.repository, tag)
+		if getReleasesErr != nil {
+			retryError = g.handleGithubErr(getReleasesErr, "failed to read release %q", tag)
+			// return immediately if we are rate limited
+			if _, ok := getReleasesErr.(*github.RateLimitError); ok {
+				return false, retryError
+			}
+			return false, nil
+		}
 
-	if release == nil {
-		return nil, errors.Errorf("failed to get release %q", tag)
+		if release == nil {
+			retryError = errors.Errorf("failed to get release %q", tag)
+			return false, nil
```

Comment:
Does it make sense to retry in this case? I assume in this case we just couldn't find a release for the tag which won't change with more retries

Reply:
I suppose the trade-off here is:

1) retry in anticipation of an unexpected empty response (something transiently pathological happening at the gh platform layer
2) blast gh with a bunch of GETs when E2E attempts to grab a release or a non-existent tag

I would guess that #2 is way more likely to happen, and so we probably shouldn't retry nil err + empty release data.

@ykakarap wdyt?

Reply:
Ah good point. I'm fine with both options.

Reply:
Good point. Actually looking at the underlying Github API it looks like `getReleasesErr` and `release` cannot both be `nil` at the same time. I believe if the release does not exists we will get back an error. So technically that piece of code is unreachable. 

How about dropping that check?

Reply:
done, thanks all!

Reply:
Good catch!